src/Task.ts

```diff
@@ -438,159 +439,78 @@ export class Task {
+        return renderTaskLine(this, renderDetails);
     }
 
     /**
-     *
-     *
+     * Flatten the task as a string that includes all its components.
      * @param {LayoutOptions} [layoutOptions]
      * @return {*}  {string}
      * @memberof Task
      */
     public toString(layoutOptions?: LayoutOptions): string {
-        layoutOptions = layoutOptions ?? new LayoutOptions();
-        let taskString = this.description;
-
-        if (!layoutOptions.hidePriority) {
-            let priority: string = '';
-
-            if (this.priority === Priority.High) {
-                priority = ' ' + prioritySymbols.High;
-            } else if (this.priority === Priority.Medium) {
-                priority = ' ' + prioritySymbols.Medium;
-            } else if (this.priority === Priority.Low) {
-                priority = ' ' + prioritySymbols.Low;
-            }
-
-            taskString += priority;
-        }
-
-        if (!layoutOptions.hideRecurrenceRule && this.recurrence) {
-            const recurrenceRule: string = layoutOptions.shortMode
-                ? ' ' + recurrenceSymbol
-                : ` ${recurrenceSymbol} ${this.recurrence.toText()}`;
-            taskString += recurrenceRule;
-        }
-
-        if (!layoutOptions.hideStartDate && this.startDate) {
-            const startDate: string = layoutOptions.shortMode
-                ? ' ' + startDateSymbol
-                : ` ${startDateSymbol} ${this.startDate.format(TaskRegularExpressions.dateFormat)}`;
-            taskString += startDate;
-        }
-
-        if (!layoutOptions.hideScheduledDate && this.scheduledDate && !this.scheduledDateIsInferred) {
-            const scheduledDate: string = layoutOptions.shortMode
-                ? ' ' + scheduledDateSymbol
-                : ` ${scheduledDateSymbol} ${this.scheduledDate.format(TaskRegularExpressions.dateFormat)}`;
-            taskString += scheduledDate;
-        }
-
-        if (!layoutOptions.hideDueDate && this.dueDate) {
-            const dueDate: string = layoutOptions.shortMode
-                ? ' ' + dueDateSymbol
-                : ` ${dueDateSymbol} ${this.dueDate.format(TaskRegularExpressions.dateFormat)}`;
-            taskString += dueDate;
+        const taskLayout = new TaskLayout(layoutOptions);
+        let taskString = '';
+        for (const component of taskLayout.layoutComponents) {
+            taskString += this.componentToString(taskLayout, component);
         }
+        return taskString;
+    }
 
-        if (!layoutOptions.hideDoneDate && this.doneDate) {
-            const doneDate: string = layoutOptions.shortMode
-                ? ' ' + doneDateSymbol
-                : ` ${doneDateSymbol} ${this.doneDate.format(TaskRegularExpressions.dateFormat)}`;
-            taskString += doneDate;
+    /**
+     * Renders a specific LayoutComponent of the task (its description, priority, etc) as a string.
+     */
+    componentToString(layout: TaskLayout, component: LayoutComponent) {
```

Comment:
Trivia: Please would you add `public` here?
Even though I don't always remember myseful, I'm trying to stick with the original policy of this project to include `public` even though it's the default.